### PR TITLE
Make BulkChange auto-closeable

### PR DESCRIPTION
**Pull request by kohsuke**
so that it can be used as below:

```
try (BulkChange bc = new BulkChange(o)) {
   ...
   bc.commit();
}
```

@reviewbybees

### CODE REVIEW

File: test/src/test/java/hudson/slaves/NodeProvisionerTest.java [hunk 1]
```diff
@@ -35,6 +35,8 @@
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+
+import static org.bouncycastle.asn1.bc.BCObjectIdentifiers.bc;
```
Comment (stephenc) [inline test/src/test/java/hudson/slaves/NodeProvisionerTest.java]: :bug:

**Comment by stephenc** [comment]
:bug: on the bouncycastle import

File: core/src/main/java/hudson/BulkChange.java [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 import hudson.model.Saveable;
 
+import java.io.Closeable;
```
Comment (jtnord) [inline core/src/main/java/hudson/BulkChange.java]: NIT: should be `java.lang.AutoCloseable`  according to the commit description

File: core/src/main/java/hudson/BulkChange.java [hunk 3]
```diff
@@ -113,6 +99,14 @@ public void commit() throws IOException {
     }
 
     /**
+     * Alias for {@link #abort()} to make {@link BulkChange} auto-closeable.
+     */
+    @Override
+    public void close() {
```
Comment (johnou) [inline core/src/main/java/hudson/BulkChange.java]: :-1: abort will be invoked every time BulkChange is used as an auto-closable resource.

Comment (johnou) [inline core/src/main/java/hudson/BulkChange.java]: ah I see, subsequent calls to close / abort will be ignored..

File: core/src/main/java/hudson/BulkChange.java (hunk 2 above)
Comment (johnou) [inline core/src/main/java/hudson/BulkChange.java]: +1

Comment (jglick) [inline core/src/main/java/hudson/BulkChange.java]: Well `Closeable extends AutoCloseable` so this works as well. I do not see any harm in implementing the stronger `Closeable`.

**Comment by jglick** [comment]
:bee:

**Pull request merged**